scene/resources/3d/primitive_meshes.cpp

```diff
@@ -324,22 +324,39 @@ Vector2 PrimitiveMesh::get_uv2_scale(Vector2 p_margin_scale) const {
 }
 
 float PrimitiveMesh::get_lightmap_texel_size() const {
-	float texel_size = GLOBAL_GET("rendering/lightmapping/primitive_meshes/texel_size");
-
 	if (texel_size <= 0.0) {
-		texel_size = 0.2;
+		return 0.2;
 	}
 
 	return texel_size;
 }
 
+void PrimitiveMesh::_on_settings_changed() {
+	float new_texel_size = float(GLOBAL_GET("rendering/lightmapping/primitive_meshes/texel_size"));
+	if (texel_size != new_texel_size) {
+		texel_size = new_texel_size;
+	}
```

Comment:
Return early if there's no change to avoid processing:

```suggestion
	if (texel_size == new_texel_size) {
		return;
	}

	texel_size = new_texel_size;
```